src/storage/recovery/recovery_manager.cpp

```diff
@@ -77,19 +79,44 @@ void RecoveryManager::ProcessCommittedTransaction(terrier::transaction::timestam
   // Begin a txn to replay changes with.
   auto *txn = txn_manager_->BeginTransaction();
 
+  // For recording recovery throughput metric
+  uint64_t elapsed_us = 0;
+  uint64_t total_elapsed_us = 0;
+  uint64_t num_txns = 0;
+  uint64_t num_bytes = 0;
+
   // Apply all buffered changes. They should all succeed. After applying we can safely delete the record
   for (uint32_t idx = 0; idx < buffered_changes_map_[txn_id].size(); idx++) {
     auto *buffered_record = buffered_changes_map_[txn_id][idx].first;
     TERRIER_ASSERT(
         buffered_record->RecordType() == LogRecordType::REDO || buffered_record->RecordType() == LogRecordType::DELETE,
         "Buffered record must be a redo or delete.");
+    {
+      common::ScopedTimer<std::chrono::microseconds> scoped_timer(&elapsed_us);
+      if (IsSpecialCaseCatalogRecord(buffered_record)) {
+        idx += ProcessSpecialCaseCatalogRecord(txn, &buffered_changes_map_[txn_id], idx);
+      } else if (buffered_record->RecordType() == LogRecordType::REDO) {
+        ReplayRedoRecord(txn, buffered_record);
+      } else {
+        ReplayDeleteRecord(txn, buffered_record);
+      }
+    }
 
-    if (IsSpecialCaseCatalogRecord(buffered_record)) {
-      idx += ProcessSpecialCaseCatalogRecord(txn, &buffered_changes_map_[txn_id], idx);
-    } else if (buffered_record->RecordType() == LogRecordType::REDO) {
-      ReplayRedoRecord(txn, buffered_record);
-    } else {
-      ReplayDeleteRecord(txn, buffered_record);
+    // Update metrics
+    total_elapsed_us += elapsed_us;
+    num_txns += 1;
```

Comment:
This is incorrect. A transaction can have multiple records, you are incrementing the txn counter once **per record** here.